server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java

```diff
@@ -35,50 +36,36 @@
   @Override
   public void kill(DataSegment segment) throws SegmentLoadingException
   {
-    final File path = getDirectory(segment);
-    log.info("segment[%s] maps to path[%s]", segment.getIdentifier(), path);
+    final File path = getPath(segment);
+    log.info("killing segment[%s] mapped to path[%s]", segment.getIdentifier(), path);
 
-    if (!path.isDirectory()) {
-      if (!path.delete()) {
-        log.error("Unable to delete file[%s].", path);
-        throw new SegmentLoadingException("Couldn't kill segment[%s]", segment.getIdentifier());
-      }
-
-      return;
-    }
+    try {
+      if (path.getName().endsWith(".zip")) {
 
-    final File[] files = path.listFiles();
-    int success = 0;
+        // path format -- > .../dataSource/interval/version/partitionNum/xxx.zip
+        File partitionNumDir = path.getParentFile();
+        FileUtils.deleteDirectory(partitionNumDir);
```

Comment:
Should we delete parent directory only when it's empty? Or at least there should be an option for it.